jboss-kie-workbench/added/launch/jboss-kie-workbench.sh

```diff
@@ -129,6 +130,37 @@ function configure_guvnor_settings() {
             JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.uberfire.nio.git.hooks=${GIT_HOOKS_DIR}"
         fi
     fi
+
+    local hostname
+    local port="80"
+    local protocol="http"
+    if [ "${WORKBENCH_ROUTE_NAME}" != "" ]; then
+
+        if [[ "${WORKBENCH_ROUTE_NAME}" = *"secure"* ]]; then
+            port="443"
+            protocol="https"
+        fi
+
+        local response=$(query_server_host ${WORKBENCH_ROUTE_NAME})
+
+        if [ "${response: -3}" = "200" ]; then
+               # parse the json response to get the route host
+               url=$(echo ${response::- 3} | python -c 'import json,sys;obj=json.load(sys.stdin);print obj["spec"]["host"]')
+               hostname="${protocol}://${url}:${port}"
+               log_info "Using route hostname: ${hostname}"
+        else
+            log_warning "Fail to query the route name using Kubernetes API, service account might not have necessary privileges, defaulting it to pod's hostname [${HOSTNAME}]."
+            if [ ! -z "${response}" ]; then
+                log_warning "Response message: ${response::- 3} - HTTP Status code: ${response: -3}"
+            fi
+            hostname="http://${HOSTNAME}:8080"
+        fi
+
+    else
+        hostname="http://${HOSTNAME}:8080"
+        log_info "Using route hostname: ${hostname}"
+    fi
+    JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.appformer.m2repo.url=http://${hostname}/kie-wb/maven2"
```

Comment:
You will have to check the $JBOSS_PRODUCT env variable, since it is "/kie-wb/" in RHPAM, but "/kie-drools-wb/" in RHDM. So you'll need an "if else". Also remember that we rename both business-central.war and decision-central.war to ROOT.war, but I'm pretty sure "/kie-*wb/ is still in the url path for both.